### PR TITLE
Update repeat deploy validation to check for sigs 

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

Supersedes https://github.com/rchain/rchain/pull/2360

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

https://rchain.atlassian.net/browse/RCHAIN-3244

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by KentShikama** [comment]
bors try

File: casper/src/test/scala/coop/rchain/casper/ValidateTest.scala [hunk 1]
```diff
@@ -391,6 +391,33 @@ class ValidateTest
...
+  }
+
+  it should "not accept blocks with a repeated deploy" in withStorage {
+    implicit blockStore => implicit blockDagStorage =>
+      for {
+        deploy  <- ConstructDeploy.basicProcessedDeploy[Task](0)
+        genesis <- createGenesis[Task](deploys = Seq(deploy))
+        block1 <- createBlock[Task](
+                   Seq(genesis.blockHash),
+                   genesis,
```
Comment (ArturGajowy) [inline casper/src/test/scala/coop/rchain/casper/ValidateTest.scala]: Why does the underlying BlockDagStore require a `genessis` parameter for insert?

```scala


trait BlockDagStorage[F[_]] {
  def getRepresentation: F[BlockDagRepresentation[F]]
  def insert(
      block: BlockMessage,
      genesis: BlockMessage,
      invalid: Boolean
  ): F[BlockDagRepresentation[F]]
```

File: casper/src/test/scala/coop/rchain/casper/ValidateTest.scala [hunk 2]
```diff
@@ -391,6 +391,33 @@ class ValidateTest
...
 
+  "Repeat deploy validation" should "return valid for empty blocks" in withStorage {
+    implicit blockStore => implicit blockDagStorage =>
+      for {
+        _      <- createChain[Task](2)
+        block  <- blockDagStorage.lookupByIdUnsafe(0)
+        block2 <- blockDagStorage.lookupByIdUnsafe(1)
+        dag    <- blockDagStorage.getRepresentation
+        _      <- Validate.repeatDeploy[Task](block, dag, 50) shouldBeF Right(Valid)
+        _      <- Validate.repeatDeploy[Task](block2, dag, 50) shouldBeF Right(Valid)
```
[the remark this answers is not in the document]
Comment (KentShikama) [reply casper/src/test/scala/coop/rchain/casper/ValidateTest.scala]: It is valid according to the repeat deploy validation: if a block doesn't have any deploys, it obviously cannot have any repeated deploys. There are other validation checks that would make it invalid overall.

`createChain` calls `createBlock` below

https://github.com/rchain/rchain/blob/0f67416bb2cdb8584fb4caf4c631b6b27eea6bb2/casper/src/test/scala/coop/rchain/casper/ValidateTest.scala#L55

which by default has no deploys

https://github.com/rchain/rchain/blob/90faa0ced3ef219d5ade464669707e49e318928e/casper/src/test/scala/coop/rchain/casper/helper/BlockGenerator.scala#L144

File: casper/src/test/scala/coop/rchain/casper/ValidateTest.scala (hunk 1 above)
Comment (KentShikama) [reply casper/src/test/scala/coop/rchain/casper/ValidateTest.scala]: Whenever we bond a new validator, their latest message needs to be set to the genesis block and so we pass it in as the BlockDagRepresentation itself isn't aware of what is the genesis block.

**Comment by KentShikama** [comment]
bors try

File: casper/src/test/scala/coop/rchain/casper/ValidateTest.scala [hunk 3]
```diff
@@ -397,6 +397,33 @@ class ValidateTest
       } yield result
   }
 
+  "Repeat deploy validation" should "return valid for empty blocks" in withStorage {
+    implicit blockStore => implicit blockDagStorage =>
+      for {
+        _      <- createChain[Task](2)
+        block  <- blockDagStorage.lookupByIdUnsafe(0)
```
Comment (tymm) [inline casper/src/test/scala/coop/rchain/casper/ValidateTest.scala]: Might help future readers if you name it `emptyBlock` and `emptyBlock2`.

**Comment by KentShikama** [comment]
bors try

**Comment by KentShikama** [comment]
bors try

**Pull request merged**